ios/browser/api/sync/brave_sync_api.mm

```diff
@@ -288,6 +364,46 @@ - (void)resetSync {
   _worker->ResetSync();
 }
 
+- (void)setJoinSyncChain:(void (^)(bool success))completion {
+  _worker->SetJoinSyncChainCallback(base::BindOnce(
+      [](void (^completion)(bool), const bool& success) {
+        completion(success);
+      },
+      completion));
```

Comment:
You shouldn't need to do this, `_worker->SetJoinSyncChainCallback(base::BindOnce(completion))` should work